miso-web/src/main/webapp/scripts/list_attachment.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+ListTarget.attachment = (function() {
+  var showUploadDialog = function(entityType, entityId) {
+    var dialogArea = jQuery('#dialog');
+    dialogArea.empty();
+
+    var form = jQuery('<form id="uploadForm">');
+    form.append(jQuery('<input id="fileInput" type="file" name="file">'));
+    dialogArea.append(form);
+
+    var dialog = dialogArea.dialog({
+      autoOpen: true,
+      width: 500,
+      title: 'Attach File',
+      modal: true,
+      buttons: {
+        upload: {
+          id: 'upload',
+          text: 'Upload',
+          click: function() {
+            if (!jQuery('#fileInput').val()) {
+              alert('No file selected!');
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of alert(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

Reply:
We've replaced most instances of `alert` with `Utils.showOkDialog` or `Utils.showConfirmDialog`

Reply:
All of those dialog functions take over the active dialog. The `LimsUtils.showDialog` function also uses `alert` in order to avoid this